codegen/src/lib.rs

```diff
@@ -425,7 +439,10 @@ impl<'a> ExampleGenerator<'a> {
             .enumerate()
             .map(|(i, type_id)| {
                 let variable_name = format!("key_{i}");
-                let type_path = type_gen.resolve_type_path(type_id).prune();
+                let type_path = type_gen
+                    .resolve_type_path(type_id)
+                    .expect("field not found")
```

Comment:
Can faulty user-provided metadata lead to this panic?

Reply:
That is a good points, I changed it now to return an error instead. I am sure there are also other places, e.g. in the subxt CLI tool or codegen, where we panic if some type ids are not valid.